corehq/apps/reports/tests/test_generic.py

```diff
@@ -74,3 +80,95 @@ def test_handles_rows(self):
         self.assertEqual(result[0], ['One', 'Two'])
         self.assertEqual(result[1], ['Three', 'Four'])
         self.assertEqual(result[2], ['Five', 'Six'])
+
+
+class GetSerializableReportParameters(TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.user = WebUser(domain='test-domain', username='test-user')
+        cls.user.save()
+        cls.addClassCleanup(cls.user.delete, deleted_by_domain='test-domain', deleted_by=None)
+        cls.report = object.__new__(get_report_class('daily_form_stats'))
+        GET_params = QueryDict('', mutable=True)
+        GET_params.update({
+            'emw': ['t__0', 't__5'],
+            'sub_time': '',
+            'startdate': '2022-03-20',
+            'enddate': '2022-03-28'
+        })
+        cls.report_params = {
+            'request': {
+                'GET': GET_params,
+                'META': {
+                    'QUERY_STRING': '',
+                    'PATH_INFO': '/a/practice-tracker/reports/export/daily_form_stats/'
+                },
+                'couch_user': cls.user._id,
+                'can_access_all_locations': True
+            },
+            'request_params': {
+                'sub_time': '',
+                'startdate': '2022-03-20T00:00:00',
+                'enddate': '2022-03-28T00:00:00'
+            },
+            'domain': 'practice-tracker',
+            'context': {}
+        }
+        cls.report.set_report_parameters(cls.report_params)
+
+    def test_returns_serializable_state(self):
+        serializable_state = self.report.get_json_report_parameters()
+        try:
+            json.dumps(serializable_state)
+        except TypeError:
+            self.fail('Expected json serializable state.')
+
+    def test_equivalent(self):
+        serializable_state = self.report.get_json_report_parameters()
+        self.assertEqual(serializable_state, self.report_params)
+
+    def test_query_dict_elements_preserved_after_serializing(self):
+        """
+        QueryDict.dict() only includes 1 element per key which is what json.dumps() invokes
+        """
+        test_querydict = QueryDict('', mutable=True)
+        test_querydict.update({'test-key': 'test-value-1'})
+        test_querydict.update({'test-key': 'test-value-2'})
+        expected_result = dict(test_querydict.lists())
+
+        self.report_params['request']['GET'] = test_querydict
+        self.report.set_report_parameters(self.report_params)
+
+        serializable_state = self.report.get_json_report_parameters()
+        serialized_state = json.dumps(serializable_state)
+        state = json.loads(serialized_state)
+        self.assertFalse(isinstance(state['request']['GET'], QueryDict))
+        self.assertEqual(expected_result, state['request']['GET'])
```

Comment:
I mentioned in another comment, but this test is probably more suitable to a serialization/deserialization utility. Reports shouldn't know the details of this.